Yoast/YoastSEO.js
# Fix tokenizer errors and tokenize < in getSentences.

## Summary

This PR can be summarized in the following changelog entry:

* Fixes assessments failing when using the less-than sign in content.

## Relevant technical choices:

* Always catch tokenizer.end() calls as errors there will severely disrupt the worker.
* Add the same regex as used in the `htmlBlockTokenizer` to the `sentenceTokenizer`.

Changes from @hansjovis :
* Refactors the `createTokenizer` and `tokenizeSentences` functions to make them reusable. These are used to tokenize the less-than sign match in a second pass. 
* Renames some local variables to make sure that they do not overshadow file-level declarations (e.g. `tokens` and `sentenceTokenizer`).
* Refactors some variables from `var` to `const` or `let`.

## Test instructions

This PR can be tested by following these steps:

* Write a beautiful post with great assessments.
* Add a less-than sign ( `<` ) to your post.
* You should still have great assessments.

For checking whether sentences are correctly found and returned, and you have access to the code, you can add these lines to `src/stringProcessing/getSentences.js` (on line 378, at the bottom of the file, before returning the sentences):
```js
console.log( sentences );
```

Fixes #840.
